src/streamlink/plugins/youtube.py

```diff
@@ -354,35 +336,32 @@ def _get_streams(self):
             log.error("Could not get video info")
             return
 
-        if info.get("livestream") == '1' or info.get("live_playback") == '1' \
-                or info.get("player_response", {}).get("videoDetails", {}).get("isLive"):
+        if (info.get("player_response", {}).get("videoDetails", {}).get("isLiveContent")
+           or info.get("player_response", {}).get("videoDetails", {}).get("isLive")):
             log.debug("This video is live.")
             is_live = True
 
-        formats = info.get("fmt_list")
         streams = {}
         protected = False
-        for stream_info in info.get("url_encoded_fmt_stream_map", []):
-            if stream_info.get("s"):
-                protected = True
-                continue
+        if (info.get("player_response", {}).get("streamingData", {}).get("adaptiveFormats", [dict()])[0].get("cipher")
+           or info.get("player_response", {}).get("streamingData", {}).get("formats", [dict()])[0].get("cipher")):
+            protected = True
+            log.debug("This video may be protected.")
 
+        for stream_info in info.get("player_response", {}).get("streamingData", {}).get("formats", []):
             stream = HTTPStream(self.session, stream_info["url"])
```

Comment:
```
$ streamlink https://www.youtube.com/watch?v=Nj2U6rhnucI
[cli][debug] OS:         Linux-5.3.0-45-generic-x86_64-with-Ubuntu-19.10-eoan
[cli][debug] Python:     2.7.15+
[cli][debug] Streamlink: 1.3.1+61.g9548a69.dirty
[cli][debug] Requests(2.22.0), Socks(1.6.7), Websocket(0.54.0)
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=Nj2U6rhnucI
[plugin.youtube][debug] Video ID from URL
[plugin.youtube][debug] Using video ID: Nj2U6rhnucI
[plugin.youtube][debug] get_video_info - 1: Found data
[plugin.youtube][debug] This video may be protected.
Traceback (most recent call last):
  File "py2/bin/streamlink", line 11, in <module>
    load_entry_point('streamlink', 'console_scripts', 'streamlink')()
  File "src/streamlink_cli/main.py", line 1050, in main
    handle_url()
  File "src/streamlink_cli/main.py", line 587, in handle_url
    streams = fetch_streams(plugin)
  File "src/streamlink_cli/main.py", line 467, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "src/streamlink/plugin/plugin.py", line 317, in streams
    ostreams = self._get_streams()
  File "src/streamlink/plugins/youtube.py", line 352, in _get_streams
    stream = HTTPStream(self.session, stream_info["url"])
KeyError: u'url'
```

same for `_create_adaptive_streams`

Reply:
Beginner's mistake on my part. 
I'm fixing it.
We should probably improve the unit test for the youtube plugin (in another PR). 
Here is a good resource (https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/extractor/youtube.py#L539)